### PR TITLE
test: read op deploy config from the `devnetL1.json` file

### PR DESCRIPTION
**Pull request by lesterli**
This PR updates the op test manager to read the op deploy config from the `devnetL1.json` file.
Close https://github.com/babylonchain/finality-provider/issues/510

## Test Plan

```
make lint
make test-e2e-op
```

### CODE REVIEW

File: itest/opstackl2/op_test_manager.go [hunk 1]
```diff
@@ -111,11 +108,12 @@ func StartOpL2ConsumerManager(t *testing.T) *OpL2ConsumerTestManager {
 	require.NoError(t, err)
 	require.Equal(t, uint64(1), opFinalityGadgetContractWasmId, "first deployed contract code_id should be 1")
 
-	// instantiate op contract
-	l2ChainID, err := jsonutil.ReadJSONValueToUint64(
-		devnetL1JsonPath, "l2ChainID")
-	require.NoError(t, err)
+	// DefaultSystemConfig load the op deploy config from devnet-data folder
+	opSysCfg := ope2e.DefaultSystemConfig(t)
+	require.Equal(t, -1, opSysCfg.DeployConfig.BabylonFinalityGadgetChainType, "should be -1 in devnetL1.json that means to connect with the Babylon localnet")
```
Comment (maurolacy) [inline itest/opstackl2/op_test_manager.go:113]: Nit: perhaps using an enum or string value here makes sense, for clarity.

Comment (bap2pecs) [reply itest/opstackl2/op_test_manager.go:113]: per runchao's suggestion, we should use chain ID. I made the changes last night. see https://github.com/babylonchain/babylon-finality-gadget/blob/main/sdk/config/config.go#L9-L12

i will update the usage here later

**Comment by bap2pecs** [comment]
changed a few more things:

- `WaitForNActiveDels` shouldn't be used. check out `WaitForDelegations`
- the test `TestFinalityStuckAndRecover` doesn't follow the design in https://github.com/babylonchain/finality-provider/issues/502. `require.Eventually` is insufficient. it's missing "wait for 5x l1_block_time"
- extract common logic to `SetupFinalityProviders`
- created helper method `getL1BlockTime()` and `getL2BlockTime()`
- format some long lines

**Comment by bap2pecs** [comment]
@lesterli  could you check whether CI fails? seems flaky though

```
=== RUN   TestFinalityStuckAndRecover
    babylon_node_handler.go:148: babylon log file: /tmp/zBabylonTest248360974/babylon.log
Starting babylond with command:  /home/circleci/.go_workspace/bin/babylond start --home=/tmp/zBabylonTest248360974/node0/babylond --log_level=trace --trace
    op_test_manager.go:81: [00:53:53.716] Fp home dir: /tmp/fpe2etest1980108828/fp-home
    eotsmanager_handler.go:23: 
                Error Trace:    /home/circleci/project/itest/eotsmanager_handler.go:23
                                                        /home/circleci/project/itest/opstackl2/op_test_manager.go:94
                                                        /home/circleci/project/itest/opstackl2/op_e2e_test.go:120
                Error:          Received unexpected error:
                                intercept already started
                Test:           TestFinalityStuckAndRecover
--- FAIL: TestFinalityStuckAndRecover (0.40s)
```

**Review by SebastianElvis — APPROVED** [review]
CI failed though

**Comment by lesterli** [comment]
> seems flaky though
indeed, run three test cases locally, all are PASS.

**Comment by lesterli** [comment]
> @lesterli could you check why CI fails? seems flaky though
> 
Checking the errors of the latest CI failed, the first failed test case throws the same error: `the EOTS manager server is not responding` mentioned by the https://github.com/babylonchain/finality-provider/issues/487

the second failed test case throws the error: `rpc error: code = NotFound desc = github.com/cosmos/cosmos-sdk/baseapp.gRPCErrorToSDKError`, it should be failed to create `NewCwClient`.

We need to add the retry logic for them.

**Comment by bap2pecs** [comment]
> We need to add the retry logic for them.

is this PR blocked by that?

**Comment by lesterli** [comment]
> is this PR blocked by that?

No, but CI could be failed by that.

**Pull request merged**